### PR TITLE
[CP-2075] Question about the passcode appears before starting the phone after the update

### PR DESCRIPTION
**Pull request by OskarMichalkiewicz**
Jira: [CP-2075]

**Description**

- [ ] getState function in async thunk actions is correctly typed
- [ ] redux selectors are used in components / prop drilling is reduce

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-2075]: https://appnroll.atlassian.net/browse/CP-2075?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/update/services/device-update.service.ts [hunk 1]
```diff
@@ -116,6 +119,19 @@ export class DeviceUpdateService {
       return deviceRestartResponse
     }
 
+    return Result.success(true)
+  }
+
+  public async checkUpdate(): Promise<ResultObject<boolean>> {
+    if (this.beforeUpdateOsVersion === null) {
```
Comment (dkarski) [inline packages/app/src/update/services/device-update.service.ts:126]: Is the lack of clarity of this variable during one of the below methods introducing any problems?


I think it's only once checked during the application runtime.

File: packages/app/src/__deprecated__/renderer/wrappers/base-app.component.tsx [hunk 2]
```diff
@@ -137,14 +137,6 @@ const BaseApp: FunctionComponent<Props> = ({
 }
 
 const isDeviceRestarting = (state: RootState & ReduxRootState): boolean => {
```
Comment (dkarski) [inline packages/app/src/__deprecated__/renderer/wrappers/base-app.component.tsx:139]: Does the problem occur with backup and restore? Do we have an issue with this in our backlog?

File: packages/app/src/update/actions/start-update-os/start-update-os.action.ts [hunk 3]
```diff
@@ -99,8 +101,12 @@ export const startUpdateOs = createAsyncThunk<
...
 
       if (release.version !== state.device.data?.osVersion) {
-        const result = await startOsUpdate({ fileName: release.file.name })
-
+        dispatch(setRestarting(true))
+        const updateResult = await startOsUpdate({
+          fileName: release.file.name,
+        })
+        dispatch(setRestarting(false))
+        const result = updateResult.ok ? await checkUpdate() : updateResult
```
Comment (dkarski) [inline packages/app/src/update/actions/start-update-os/start-update-os.action.ts:109]: Well done! 👌

File: packages/app/src/__deprecated__/renderer/wrappers/base-app.component.tsx (hunk 2 above)
Comment (OskarMichalkiewicz) [reply packages/app/src/__deprecated__/renderer/wrappers/base-app.component.tsx:139]: no, and no :)

Comment (dkarski) [reply packages/app/src/__deprecated__/renderer/wrappers/base-app.component.tsx:139]: it's strange that we have a problem with update but not with other processes 🤔

Comment (dkarski) [reply packages/app/src/__deprecated__/renderer/wrappers/base-app.component.tsx:139]: Thank you for checking it out <3

**Review by dkarski — APPROVED** [review]

**Review by lkowalczyk87 — APPROVED** [review]

**Pull request merged**